firmware/arduino/chessboard/motorDriver.ino

```diff
@@ -83,32 +83,26 @@ void home()
 // Moves the magnet from the "start" point to the "end" point
 // This can only move in straight lines
 // Returns a boolean indicating success/error
-bool moveStraight(int motor[], float startCol, float startRow, float endCol, float endRow)
+bool moveStraight(int motor[], int startCol, int startRow, int endCol, int endRow)
 {
   // A specific motor is passed to this function since we are only moving one here
 
   // How many steps per space
-  float pieceSpaces;
-  int dir, numSteps;
+  int dir, numSteps, unitSpaces;
   int i;
 
-  // Makes sure all arugments produce a multiple of unitspaces
-  if (fmodf(startCol, 0.5) != 0 || fmodf(startRow, 0.5) != 0 ||
-      fmodf(endCol, 0.5) != 0 || fmodf(endRow, 0.5) != 0)
-    return false;
-
   // This could be two cases, x or y movement
   if (endRow == startRow)
   {
     // X movement
-    pieceSpaces = fabs(endCol - startCol);
+    unitSpaces = abs(endCol - startCol);
     dir = (endCol > startCol) ? RIGHT : LEFT;
     setScale(xMotor, WHOLE_STEPS);
   }
   else if (endCol == startCol)
   {
     // Y movement
-    pieceSpaces = fabs(endRow - startRow);
+    unitSpaces = abs(endRow - startRow);
```

Comment:
On line 147 you have a comment about abs and how it's used, but not here. If you can make sure that "like" code blocks have the same comments, assuming they're applicable, please do so. It increases readability and consistency.